1st seminar 코드 리뷰 해주세요😄

yjooooo opened the pull request:
풀리퀘를 모르고 닫아버려서 새로 열었어요~! 여기에 코드 리뷰 해주시면 감사하겠습니다!!

yjooooo committed:
update 1st seminar

yjooooo committed:
Add 2nd week content

yjooooo committed:
#1 [UI] 1st Seminar : Make Layout
- activity_sign_in.xml
- activity_sign_up.xml
- activity_home.xml

yjooooo committed:
#2 [CHORE] 1st Seminar : addktlintcheckgitprecommithook

yjooooo committed:
#3 [FEAT] 1st Seminar : Level 1 Clear
- SignInActivity 동작 구현
- SignUpActivity 동작 구현
- SignUpActivity의 정보를 SignInActivity로 가져오기

yjooooo committed:
#5 [DOC] 1st Seminar : Upload README.md

yjooooo committed:
#6 [FEAT] 1st Seminar : Add LifeCycle Log

yjooooo committed:
#8 [MODIFY] 1st Seminar : Modify LifeCycle Log Message, Edit README

yjooooo committed:
[DOC] Update README.md

yjooooo committed:
#11 [MODIFY] 1st Seminar : 디자인 추가작업, 패키지정리

yjooooo committed:
#12 [CHORE] 1st Seminar : Modify .gitignore

yjooooo committed:
Update README.md

yjooooo committed:
[MODIFY] Modify Something.
- Delete blank check method
- Make ToastMessage util method

l2hyunwoo commented on app/build.gradle:
```diff
@@ -39,7 +43,13 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+
+    // DataBinding
+    kapt 'com.android.databinding:compiler:6.5'
```
음 나 이건 처음봤는데 나도 적용해봐야겠다 감사

l2hyunwoo commented on app/src/main/java/com/yjooooo/sopt28th/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,87 @@
+    private fun setOnLoginBtnClick() {
+        binding.signInBtnLogin.setOnClickListener {
+            if (isUserInfoNotBlank()) {
+                ToastMessageUtil(this, binding.signInEdtId.text.toString() + "님 로그인되었습니다.")
```
```suggestion
fun Context.ToastMessageUtil(message: String) {
    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
}
ToastMessageUtil(binding.signInEdtId.text.toString() + "님 로그인되었습니다.")
```

이런 식으로 만들면 매개변수를 하나 더 줄일 수 있지 않을까?

l2hyunwoo commented on app/src/main/java/com/yjooooo/sopt28th/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,87 @@
+
+    private fun setIdPwAfterSignUp(activityResult: ActivityResult) {
+        if (activityResult.resultCode == RESULT_OK) {
+            with(binding) {
+                signInEdtId.setText(activityResult.data!!.getStringExtra("id"))
+                signInEdtPw.setText(activityResult.data!!.getStringExtra("pw"))
+            }
```
GOOD! 역시 손차장 기능분리 아주 철저하게 해주는구만

l2hyunwoo commented on app/src/main/java/com/yjooooo/sopt28th/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,87 @@
+        }
+    }
+
+    private fun isUserInfoNotBlank(): Boolean {
+        with(binding) {
+            return signInEdtId.text.isNotBlank() && signInEdtPw.text.isNotBlank()
+        }
+    }
```
```suggestion
    private fun isUserInfoNotBlank() =
            binding.signInEdtId.text.isNotBlank() && binding.signInEdtPw.text.isNotBlank()
```

한줄로 적을 수 있는 함수는 그냥 한 줄로 적는게 조금 더 깔끔한 것 같아

l2hyunwoo commented on app/src/main/java/com/yjooooo/sopt28th/ui/signup/SignUpActivity.kt:
```diff
@@ -0,0 +1,67 @@
+        Log.d("LifeCycle", "SignUp_onDestroy")
+    }
+
+    private fun setOnSignUpBtnClick() {
```
와 함수 이름 좋습니다....

l2hyunwoo commented on build.gradle:
```diff
@@ -1,14 +1,39 @@
+apply plugin: "org.jlleitschuh.gradle.ktlint"
+
+ktlint {
+    version = "0.34.2"
```
ktlint 버전 올리자! 0.41.0가지 나왔을거야

yjooooo replied:
오케이!

l2hyunwoo commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,17 @@
 <resources>
-    <string name="app_name">28th-SOPT</string>
+    <string name="app_name">My GitHuby</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="app_logo">app logo</string>
+    <string name="id_hint">ID를 입력해주세요</string>
+    <string name="pw_hint">비밀번호를 입력해주세요</string>
+    <string name="name_hint">이름 입력해주세요</string>
+    <string name="sign_in_btn_login">login</string>
+    <string name="sign_in_txt_no_id">아이디/비밀번호가 없으신가요?</string>
+    <string name="sign_in_txt_sign_up">회원가입</string>
+    <string name="sign_up_btn_sign_up">signup</string>
+    <string name="sign_up_txt_name">이름</string>
+    <string name="sign_up_txt_id">Github 아이디</string>
+    <string name="sign_up_txt_pw">비밀번호</string>
+    <string name="sign_in_txt_id">Github 아이디</string>
+    <string name="sign_in_txt_pw">비밀번호</string>
```
아 아주 좋아요 이런 습관

l2hyunwoo reviewed (approved):
27기의 손연주와 28기의 손연주는 진짜 완전 다르구나....이제 하산하십쇼

yjooooo committed:
[MODIFY] Modify Code After Code Review
- Rename ToastmessageUtil
- Modify Ktlint Version
- Modify ToastMessageUtil Parameter

jinsu4755 commented on app/src/main/java/com/yjooooo/sopt28th/util/toastMessageUtil.kt:
```diff
@@ -0,0 +1,8 @@
+import android.content.Context
+import android.widget.Toast
+
+fun Context.toastMessageUtil(text: String) {
```
세상에 이름 깔쌈한거 보소 ㄷㄷㄷㄷ 잘했다👍👍

jinsu4755 commented on app/src/main/java/com/yjooooo/sopt28th/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,85 @@
+
+    private fun setIdPwAfterSignUp(activityResult: ActivityResult) {
+        if (activityResult.resultCode == RESULT_OK) {
+            with(binding) {
```
with 사용 굳굳!

yjooooo replied:
고마워유~ 뭔가 함수 이름 맘에 안들어서 setUserInfoAfterSignup으로 바꿨다 ㅋㅋㅋㅋ

onpyeong replied:
오 이렇게도 사용할 수 있구나!

jinsu4755 reviewed (commented):
너무 잘했구만!! 굳굳 최고다!

yjooooo committed:
#14 [MODIFY] 1st Seminar : Rename Method
